cvmfs/network/sink_file.h

```diff
@@ -0,0 +1,113 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_NETWORK_SINK_FILE_H_
+#define CVMFS_NETWORK_SINK_FILE_H_
+
+#include <cerrno>
+#include <cstdio>
+#include <string>
+
+#include "sink.h"
+#include "util/posix.h"
+
+namespace cvmfs {
+
+class FileSink : public Sink {
+ public:
+  explicit FileSink(FILE *destination_file) : Sink(true),
```

Comment:
I think usually we don't own the FILE pointer, except when used from the path sink. So we should have an `Adopt` call, similar to the memory sink.

```suggestion
  explicit FileSink(FILE *destination_file) : Sink(false),
```